pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/multifile/ApexMultifileVisitorFacade.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.multifile;
+
+import net.sourceforge.pmd.lang.apex.ast.ApexNode;
+import net.sourceforge.pmd.lang.apex.ast.ApexParserVisitorAdapter;
+
+/**
+ * @author Clément Fournier
+ */
+public class ApexMultifileVisitorFacade extends ApexParserVisitorAdapter {
+
+    public void initializeWith(ApexNode<?> rootNode) {
+        ApexMultifileVisitor visitor = new ApexMultifileVisitor(ApexProjectMirror.INSTANCE);
```

Comment:
I'm still thinking about the static shared instance here. It makes it difficult to unit test PMD, since one test could affect a test running later.

I'm wondering, whether we could enhance the LanguageHandler implementations somehow:
* Assuming, that we reuse the same instance of the languagehandler, we could allow the implementations to do a bit more than just returning different visitors/facades - e.g. returning the same instance of the facade.
But looking at the current code, this might not work out of the box
e.g. for java, each version uses a separate instance of the language handler, but: for each PMD run, theoretically only one version of a language is active/used (either the default or the one given as command line parameter). There is still the possibility that multiple different languages are used during one PMD run, but that's not a problem.
* We could then e.g. have a reset method on the language handler?

Reply:
So you'd have a ProjectMirror as an instance variable of the LanguageVersionHandler? I like the idea, it makes more sense than keeping a static variable. We could then reset the language handler in the test runner before each test I guess

I'll do that in this PR if you'd like